test/rage2e/rag_test.go

```diff
@@ -0,0 +1,340 @@
+var modelInfo map[string]string
+var azureClusterName string
+
+var _ = Describe("RAGEngine", func() {
+	BeforeEach(func() {
+		loadTestEnvVars()
+		loadModelVersions()
+	})
+
+	AfterEach(func() {
+		if CurrentSpecReport().Failed() {
+			utils.PrintPodLogsOnFailure(namespaceName, "")     // The Preset Pod
+			utils.PrintPodLogsOnFailure("kaito-ragengine", "") // The Kaito Workspace Pod
+			utils.PrintPodLogsOnFailure("kaito-ragengine", "") // The Kaito ragengine Pod
+			utils.PrintPodLogsOnFailure("gpu-provisioner", "") // The gpu-provisioner Pod
+			Fail("Fail threshold reached")
+		}
+	})
+
+	It("should create RAG with localembedding and huggingface API successfully", func() {
+		numOfNode := 1
+
+		ragengineObj := createLocalEmbeddingHFURLRAGEngine()
+
+		defer cleanupResources(nil, ragengineObj)
+
+		validateRAGEngineResourceStatus(ragengineObj)
+		validateAssociatedService(ragengineObj.ObjectMeta)
+		validateInferenceandRAGResource(ragengineObj.ObjectMeta, int32(numOfNode), false)
+		validateRAGEngineReadiness(ragengineObj)
+
+	})
+})
+
+func createAndValidateRAGEngine(ragEngineObj *kaitov1alpha1.RAGEngine) {
+	By("Creating ragEngine", func() {
+		Eventually(func() error {
+			return utils.TestingCluster.KubeClient.Create(ctx, ragEngineObj, &client.CreateOptions{})
+		}, utils.PollTimeout, utils.PollInterval).
+			Should(Succeed(), "Failed to create ragEngine   %s", ragEngineObj.Name)
+
+		By("Validating ragEngine creation", func() {
+			err := utils.TestingCluster.KubeClient.Get(ctx, client.ObjectKey{
+				Namespace: ragEngineObj.Namespace,
+				Name:      ragEngineObj.Name,
+			}, ragEngineObj, &client.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+}
+
+func GenerateLocalEmbeddingRAGEngineManifest(name, namespace, instanceType, embeddingModelID string, labelSelector *metav1.LabelSelector, inferenceURL *kaitov1alpha1.InferenceServiceSpec) *kaitov1alpha1.RAGEngine {
+	return &kaitov1alpha1.RAGEngine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: &kaitov1alpha1.RAGEngineSpec{
+			Compute: &kaitov1alpha1.ResourceSpec{
+				InstanceType:  instanceType,
+				LabelSelector: labelSelector,
+			},
+			Embedding: &kaitov1alpha1.EmbeddingSpec{
+				Local: &kaitov1alpha1.LocalEmbeddingSpec{
+					ModelID: embeddingModelID,
+				},
+			},
+			InferenceService: inferenceURL,
+		},
+	}
+}
+
+func createLocalEmbeddingHFURLRAGEngine() *kaitov1alpha1.RAGEngine {
+	ragEngineObj := &kaitov1alpha1.RAGEngine{}
+	hfURL := "https://api-inference.huggingface.co/models/HuggingFaceH4/zephyr-7b-beta/v1/completions"
+	By("Creating RAG with localembedding and huggingface API", func() {
+		uniqueID := fmt.Sprint("rag-", rand.Intn(1000))
+		ragEngineObj = GenerateLocalEmbeddingRAGEngineManifest(uniqueID, namespaceName, "Standard_NC6s_v3", "BAAI/bge-small-en-v1.5",
+			&metav1.LabelSelector{
+				MatchLabels: map[string]string{"apps": "phi-3"},
+			},
+			&kaitov1alpha1.InferenceServiceSpec{
```

Comment:
If it is hf endpoint, you would need to have a secret to save the access token, right?

Reply:
we don't need token for this one